src/main/python/pybuilder/reactor.py

```diff
@@ -90,7 +90,9 @@ def validate_project(self):
     def prepare_build(self,
                       property_overrides=None,
                       project_directory=".",
-                      project_descriptor="build.py"):
+                      project_descriptor="build.py",
+                      exclude_optional_tasks=[],
```

Comment:
mutable kwargs are bad practice... the `if X is None` pattern sucks, but it's safer in case we accidentally mutate the lists